### PR TITLE
feat: fix duplicate uploaded key

### LINKED ISSUE #1352: Dev Keys - Upload same key error
**CONTEXT**

We get an 'Internal Server Error' for Uploading same Developer Keys. Only after Revoking the original key, you can upload the key again.

**Steps to reproduce**: 
- Go to Settings/Developer Keys
- Upload a key in the Developer Keys section for a Payment Pointer
- Try to upload the same key again to the same PP or a different one.

**Tasks**
- [ ] check if it is intended behaaviour to have unique Developer Keys
- [ ] check if one Payment Pointer can't have the same key multiple times, but different Payment Pointers can have the same Developer Key
- [ ] update the code accordingly to the above findings, with either fixing Internal Server Error, or appropiate Error Messages that can be displayed for the user

### PR DESCRIPTION
**Pull request by rico191013**
### Context

- fixes #1352

### CODE REVIEW

File: packages/wallet/backend/src/walletAddressKeys/service.ts [hunk 1]
```diff
@@ -91,7 +92,15 @@ export class WalletAddressKeyService implements IWalletAddressKeyService {
       walletAddressId
     }
 
-    await WalletAddressKeys.query().insert(key)
+    try {
+      await WalletAddressKeys.query().insert(key)
+    } catch (e) {
+      if (e instanceof UniqueViolationError)
+        throw new BadRequest(
+          'Same key already uploaded. Please upload a unique one'
```
Comment (Tymmmy) [inline packages/wallet/backend/src/walletAddressKeys/service.ts:100]:
```suggestion
          'Same key already uploaded. Please upload a unique one.'
```

**Comment by raducristianpopa** [comment]
If the users have the same key (by any chance), will they be able to upload the same one?

**Review by Tymmmy — APPROVED** [review]

**Comment by Tymmmy** [comment]
@rico191013 please take a look at Radu's question and also on the story check the items that you solved.

File: packages/wallet/backend/src/walletAddressKeys/service.ts [hunk 2]
```diff
@@ -91,7 +92,15 @@ export class WalletAddressKeyService implements IWalletAddressKeyService {
       walletAddressId
     }
 
-    await WalletAddressKeys.query().insert(key)
+    try {
+      await WalletAddressKeys.query().insert(key)
```
Comment (rico191013) [inline packages/wallet/backend/src/walletAddressKeys/service.ts:96]: @raducristianpopa For the sake of having a unique ID for each key, every user can have one unique key for uploading, ensuring that others can't upload the same key.

**Pull request merged**